genesyscloud/data_source_genesyscloud_knowledge_category.go

```diff
@@ -0,0 +1,98 @@
+package genesyscloud
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v94/platformclientv2"
+)
+
+func dataSourceKnowledgeCategory() *schema.Resource {
+	return &schema.Resource{
+		Description: "Data source for Genesys Cloud Knowledge Base Category. Select a category by name.",
+		ReadContext: readWithPooledClient(dataSourceKnowledgeCategoryRead),
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Knowledge base category name",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"knowledge_base_name": {
+				Description: "Knowledge base name",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func dataSourceKnowledgeCategoryRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	sdkConfig := m.(*ProviderMeta).ClientConfig
+	knowledgeAPI := platformclientv2.NewKnowledgeApiWithConfig(sdkConfig)
+
+	name := d.Get("name").(string)
+	knowledgeBaseName := d.Get("knowledge_base_name").(string)
+
+	return withRetries(ctx, 15*time.Second, func() *resource.RetryError {
+		for pageNum := 1; ; pageNum++ {
+			const pageSize = 100
+			publishedKnowledgeBases, _, getPublishedErr := knowledgeAPI.GetKnowledgeKnowledgebases("", "", "", fmt.Sprintf("%v", pageSize), knowledgeBaseName, "", true, "", "")
+			unpublishedKnowledgeBases, _, getUnpublishedErr := knowledgeAPI.GetKnowledgeKnowledgebases("", "", "", fmt.Sprintf("%v", pageSize), knowledgeBaseName, "", false, "", "")
+
+			if getPublishedErr != nil {
+				return resource.NonRetryableError(fmt.Errorf("Failed to get knowledge base %s: %s", knowledgeBaseName, getPublishedErr))
+			}
+			if getUnpublishedErr != nil {
+				return resource.NonRetryableError(fmt.Errorf("Failed to get knowledge base %s: %s", knowledgeBaseName, getPublishedErr))
```

Comment:
Should you be returning `getUnpublishedErr` here?

Reply:
Yes, thanks.